test/lib/ufe/testUfePythonImport.py

```diff
@@ -0,0 +1,39 @@
+#!/usr/bin/env python
+
+#
+# Copyright 2020 Autodesk
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import unittest
+
+
+class UfePythonImportTestCase(unittest.TestCase):
```

Comment:
This test is failing on windows:
```5: ======================================================================
5: ERROR: testImportModule (testUfePythonImport.UfePythonImportTestCase)
5: ----------------------------------------------------------------------
5: Traceback (most recent call last):
5:   File "testUfePythonImport.py", line 35, in testImportModule
5:     invalidPrim = mayaUsdUfe.getPrimFromRawItem(0)
5: TypeError: No to_python (by-value) converter found for C++ type: class pxrInternal_v0_20__pxrReserved__::UsdPrim
5:
5: ----------------------------------------------------------------------```

Reply:
Hmm... That's the exact issue that these changes are meant to address. It's a result of the ufe library being loaded *without* having loaded the usd library. That's what the script module loader function in moduleDeps.cpp should fix.

I still don't have access to a Windows machine on my end, so I'm not able to test this myself at the moment. Could I have you run the test with the environment variable `TF_DEBUG=TF_SCRIPT_MODULE_LOADER` set so we can verify whether or not the module is getting registered?

Reply:
I did a quick test of this on Linux setting `TF_DEBUG=TF_SCRIPT_MODULE_LOADER` in an interactive session of Maya after building this branch.

I ran this command in the script editor:
```from mayaUsd import ufe as mayaUsdUfe```

And I see this output in the terminal:
```
Registering library tf with predecessors: arch, 
Registering library trace with predecessors: arch, js, tf, 
Registering library gf with predecessors: arch, tf, 
Registering library vt with predecessors: arch, gf, tf, trace, 
Registering library work with predecessors: tf, trace, 
Registering library plug with predecessors: arch, js, tf, trace, work, 
Registering library ar with predecessors: arch, js, plug, tf, vt, 
Registering library sdf with predecessors: ar, arch, gf, tf, trace, vt, work, 
Registering library pcp with predecessors: ar, sdf, tf, trace, vt, work, 
Registering library kind with predecessors: plug, tf, 
Registering library usd with predecessors: ar, arch, kind, pcp, plug, sdf, tf, trace, vt, work, 
Registering library usdGeom with predecessors: js, plug, sdf, tf, trace, usd, vt, work, 
Registering library usdUtils with predecessors: arch, gf, sdf, tf, usd, usdGeom, 
Registering library usdSkel with predecessors: arch, gf, sdf, tf, trace, usd, usdGeom, vt, work, 
Registering library ndr with predecessors: ar, plug, sdf, tf, vt, work, 
Registering library sdr with predecessors: ar, ndr, sdf, tf, vt, 
Registering library usdShade with predecessors: ndr, sdf, sdr, tf, usd, usdGeom, vt, 
Registering library usdLux with predecessors: sdf, tf, usd, usdGeom, vt, 
Registering library usdRi with predecessors: sdf, tf, usd, usdGeom, usdLux, usdShade, vt, 
Registering library usdHydra with predecessors: tf, usd, usdShade, 
Registering library garch with predecessors: arch, tf, 
Registering library glf with predecessors: ar, arch, garch, gf, hf, js, plug, sdf, tf, trace, 
Registering library pxOsd with predecessors: gf, tf, vt, 
Registering library usdVol with predecessors: js, plug, sdf, tf, trace, usd, usdGeom, vt, work, 
Registering library cameraUtil with predecessors: gf, tf, 
Registering library usdImagingGL with predecessors: ar, garch, gf, glf, hd, hdx, hio, plug, pxOsd, sdf, sdr, tf, trace, usd, usdGeom, usdHydra, usdImaging, usdShade, vt, work, 
Registering library mayaUsd with predecessors: ar, gf, hd, hdx, js, kind, plug, sdf, tf, usd, usdGeom, usdImaging, usdImagingGL, usdLux, usdRi, usdShade, usdSkel, usdUtils, vt, 
Registering library ufe with predecessors: mayaUsd, sdf, tf, usd, usdGeom, vt, 
_LoadUpTo('ufe') {
  Load('tf');
  _LoadUpTo('tf') {
  }
  Load('trace');
  _LoadUpTo('trace') {
  }
  Load('work');
  _LoadUpTo('work') {
  }
  Load('plug');
  _LoadUpTo('plug') {
  }
  Load('gf');
  _LoadUpTo('gf') {
  }
  Load('vt');
  _LoadUpTo('vt') {
  }
  Load('ar');
  _LoadUpTo('ar') {
  }
  Load('kind');
  _LoadUpTo('kind') {
  }
  Load('sdf');
  _LoadUpTo('sdf') {
  }
  Load('pcp');
  _LoadUpTo('pcp') {
  }
  Load('usd');
  _LoadUpTo('usd') {
  }
  Load('usdGeom');
  _LoadUpTo('usdGeom') {
  }
  Load('garch');
  _LoadUpTo('garch') {
  }
  Load('glf');
  _LoadUpTo('glf') {
  }
  Load('pxOsd');
  _LoadUpTo('pxOsd') {
  }
  Load('ndr');
  _LoadUpTo('ndr') {
  }
  Load('sdr');
  _LoadUpTo('sdr') {
  }
  Load('usdShade');
  _LoadUpTo('usdShade') {
  }
  Load('usdHydra');
  _LoadUpTo('usdHydra') {
  }
  Load('usdImagingGL');
  _LoadUpTo('usdImagingGL') {
  }
  Load('usdLux');
  _LoadUpTo('usdLux') {
  }
  Load('usdRi');
  _LoadUpTo('usdRi') {
  }
  Load('usdSkel');
  _LoadUpTo('usdSkel') {
  }
  Load('usdUtils');
  _LoadUpTo('usdUtils') {
  }
  Load('mayaUsd');
  _LoadUpTo('mayaUsd') {
  }
}
```

I guess I'd hope to see something similar on Windows.

Reply:
```
5:  MAYA_NO_STANDALONE_ATEXIT=1
5:  TF_DEBUG=TF_SCRIPT_MODULE_LOADER
5: Test timeout computed to be: 10000000
5: MAYA_DEBUG_NO_SAVE_ON_CRASH is set.
5: Warning: At initialization event does not appear in the list of available events
5: Warning: At initialization event does not appear in the list of available events
5: Warning: At initialization event does not appear in the list of available events
5: Warning: At initialization event does not appear in the list of available events
5: Registering library tf with predecessors: arch,
5: Registering library work with predecessors: tf, trace,
5: Registering library trace with predecessors: arch, js, tf,
5: Registering library plug with predecessors: arch, js, tf, trace, work,
5: Registering library kind with predecessors: plug, tf,
5: Registering library gf with predecessors: arch, tf,
5: Registering library vt with predecessors: arch, gf, tf, trace,
5: Registering library ar with predecessors: arch, js, plug, tf, vt,
5: Registering library sdf with predecessors: ar, arch, gf, tf, trace, vt, work,
5: Registering library pcp with predecessors: ar, sdf, tf, trace, vt, work,
5: Registering library usd with predecessors: ar, arch, kind, pcp, plug, sdf, tf, trace, vt, work,
5: *** Not loading modules for unknown lib 'ufe'
5: _LoadUpTo('tf') {
5: }
5: E
5: ======================================================================
5: ERROR: testImportModule (testUfePythonImport.UfePythonImportTestCase)
5: ----------------------------------------------------------------------
5: Traceback (most recent call last):
5:   File "testUfePythonImport.py", line 35, in testImportModule
5:     invalidPrim = mayaUsdUfe.getPrimFromRawItem(0)
5: TypeError: No to_python (by-value) converter found for C++ type: class pxrInternal_v0_20__pxrReserved__::UsdPrim
5:
5: ----------------------------------------------------------------------
5: Ran 1 test in 0.184s
5:
5: FAILED (errors=1)
1/1 Test #5: testUfePythonImport ..............***Failed   12.06 sec
```

Reply:
Yeah, that seems suspicious. I would expect to see "Registering library ..." lines for both mayaUsd and ufe listed there, and there is also a big chunk of USD libraries missing as well.

Does the `testMayaUsdPythonImport` test pass for you using that build? I would expect not given what we're seeing here. In #207 where we did this for mayaUsd, I know @HamedSabri-adsk ran into an issue with the equivalent mayaUsd test whereas @fowlertADSK did not, so I wonder if something similar is happening here.

Reply:
`testMayaUsdPythonImport` is passing.

Reply:
@mattyjams, I also get the exact behavior on Windows as Krystian.

Reply:
Cool, thanks for trying it @HamedSabri-adsk!

I have a hunch that it's something to do with the fact that unlike mayaUsd, there's no actual "ufe" library that gets installed (on Linux, no "lib/libufe.so"). The UFE functionality is built as just more symbols in libmayaUsd.so. I'm not sure why that is problematic only on Windows, and I'm unfortunately not able to tinker with it on that platform at the moment. :(